src/main/java/com/es/phoneshop/model/dao/ArrayListProductDao.java

```diff
@@ -19,10 +22,13 @@ public ArrayListProductDao() {
     @Override
     public Product getProduct(Long id) throws ProductNotFoundException {
         synchronized (lock) {
-            return products.stream().
-                    filter(product -> id.equals(product.getId()))
+            if(id == null) {
+                throw new ProductNotFoundException("Given id is null when using getProduct");
```

Comment:
For wrong method parameter value you should throw IllegalArgumentException instead of ProductNotFoundException